pyscript.core/src/stdlib/pyweb/pydom.py

```diff
@@ -0,0 +1,420 @@
+            element.html = html
+
+        if label is not None:
+            element.label = label
+
+        if is_child:
+            self.append(element)
+
+        return element
+
+    # -------- Boilerplate Proxy for the Element API -------- #
+    @property
+    def appendChild(self):
+        return self._element.appendChild
+
+
+class Element(BaseElement):
+    def append(self, child):
+        # TODO: this is Pyodide specific for now!!!!!!
+        # if we get passed a JSProxy Element directly we just map it to the
+        # higher level Python element
+        if isinstance(child, JsProxy):
+            return self.append(self.from_js(child))
+
+        elif isinstance(child, Element):
+            self.appendChild(child._element)
+
+            return child
+
+    def from_js(self, js_element):
+        return self.__class__(js.tagName, parent=self)
+
+    def query(self, selector):
+        """The querySelector() method of the Element interface returns the first
+        element that is a descendant of the element on which it is invoked that
+        matches the specified group of selectors.
+        """
+        return self.__class__(self._element.querySelector(selector))
+
+    def query_all(self, selector):
+        """The querySelectorAll() method of the Element interface returns a static
+        (not live) NodeList representing a list of the document's elements that
+        match the specified group of selectors.
+        """
+        for element in self._element.querySelectorAll(selector):
+            yield self.__class__(element)
+
+    # -------- Boilerplate Proxy for the Element API -------- #
+    @property
+    def html(self):
+        return self._element.innerHTML
+
+    @html.setter
+    def html(self, value):
+        self._element.innerHTML = value
+
+    @property
+    def content(self):
+        return self._element.innerHTML
+
+    @content.setter
+    def content(self, value):
+        display(value, target=self.id)
+
+    @property
+    def id(self):
+        return self._element.id
+
+    @id.setter
+    def id(self, value):
+        self._element.id = value
+
+    @property
+    def checked(self):
+        return self._element.checked
+
+    @checked.setter
+    def checked(self, value):
+        self._element.checked = value
+
+    @property
+    def value(self):
+        tag = self._element.tagName
+        if tag == "INPUT":
+            if self._element.type == "checkbox":
+                return self._element.checked
+            elif self._element.type == "number":
+                return float(self._element.value)
+            else:
+                return self._element.value
+        return self._element.innerHTML
+
+    @value.setter
+    def value(self, value):
+        # TODO: This needs a bit more thinking. SHould we set .innerHTML or .text for instance?
+        tag = self._element.tagName
+        # print(f"Writing ({tag} )---> {self._selector} ---> {value}")
+        if tag == "INPUT":
```

Comment:
see my previous comment about reducing the HTML to an INPUT when it comes to `value` accessor.